src/helper/segment_info_parser.py

```diff
@@ -55,65 +40,71 @@ def parse_segment_words_to_dict(words_array):  # type words_array: [[]] -> [dict
     if words_array is None:
         return new_word_array
     for word_array in words_array:
-        if not isinstance(word_array[2], str):
+        if not isinstance(word_array.word, str):
             continue  # Skip if word is not a string
         word_dict = {
```

Comment:
is word_array of type dataclass? since if that is why dataclass was imported you could do this as a one liner as

```py
for word_array in words_array:
    word_dict = {key: value for key, value in asdict(word_array).items() if key in ["start", "end", "word", "probability"]}
```
or
```py
for word_array in words_array:
    word_dict = {key: getattr(word_array, key) for key in ["start", "end", "word", "probability"]}
```

is this more readable?
Probably not
But
BUT!